src/cpp/benders/benders_by_batch/BendersByBatch.cpp

```diff
@@ -151,6 +156,9 @@ void BendersByBatch::SeparationLoop() {
     SolveBatches();
 
     if (Rank() == rank_0) {
+      outer_loop_criterion_.push_back(_data.outer_loop_current_iteration_data.outer_loop_criterion);
```

Comment:
Hmm when doing Benders by Batch the criterion is likely to be irrelevant, except for the last iteration where all batches have been solved, therefore I think in this case only the strategy with criterion computation at the end of Benders should be implemented, keep it like this for this PR, change it later

Reply:
But still, information about partial criterion values from subproblems may be relevant, reflexion is needed on the design and on what information to store

Reply:
i'll comment theses lines #TODO